pkg/handlers/v1/notification.go

```diff
@@ -0,0 +1,71 @@
+package v1
+
+import (
+	"context"
+
+	"github.com/asecurityteam/nexpose-scan-notifier/pkg/domain"
+	"github.com/asecurityteam/nexpose-scan-notifier/pkg/logs"
+)
+
+// Output contains a list of completed Nexpose scans.
+type Output struct {
+	Response []scanNotification `json:"response"`
+}
+
+// scanNotification represents
+type scanNotification struct {
+	ScanID string `json:"scanID"`
+	SiteID string `json:"siteID"`
+}
+
+// NotificationHandler takes a duration and returns a list of completed scans.
+type NotificationHandler struct {
+	ScanFetcher      domain.ScanFetcher
+	TimestampFetcher domain.TimestampFetcher
+	TimestampStorer  domain.TimestampStorer
+	Producer         domain.Producer
+	LogFn            domain.LogFn
+}
+
+// Handle queries for completed scans since the last invocation, produces all
+// completed scans to a queue, and returns the list of completed scans.
+func (h *NotificationHandler) Handle(ctx context.Context) (Output, error) {
+	logger := h.LogFn(ctx)
+
+	lastScanTimestamp, err := h.TimestampFetcher.FetchTimestamp(ctx)
+	switch err.(type) {
+	case nil:
+	case domain.TimestampNotFound:
+	default:
+		logger.Error(logs.StorageFailure{Reason: err.Error()})
+		return Output{}, err
+	}
+
+	scans, err := h.ScanFetcher.FetchScans(ctx, lastScanTimestamp)
+	if err != nil {
+		logger.Error(logs.ScanFetcherFailure{Reason: err.Error()})
+		return Output{}, err
+	}
+
+	scanNotifications := make([]scanNotification, len(scans))
+	for i, scan := range scans {
```

Comment:
nit: I sometimes have trouble distinguishing between `i`, `l`, and `1`. Could we use `idx` or `index` or `offset` or something else that disambiguates the chars?

Reply:
Fixed. Renamed this to `offset`.